tests/abortStatus/enc/enc.cpp

```diff
@@ -5,85 +5,56 @@
 #include <openenclave/enclave.h>
 #include <openenclave/internal/atexit.h>
 #include <openenclave/internal/print.h>
-#include "../args.h"
+#include <atomic>
+#include "abortStatus_t.h"
 
 // Explicitly call oe_abort to abort the enclave.
-OE_ECALL void RegularAbort(void* args_)
+extern "C" int regular_abort(void)
```

Comment:
I don't think you need the extern "C"

Reply:
**extern "C"** is necessary.  This method is called from the edger8r generated file **abortStatus_t.c**.  That is a C file which will be looking for C symbols.  The **extern "C"** is needed to generate C symbols for a function that is compiled in a C++ source file.

Reply:
abortStatus_t.h declares the edge functions within an extern "C" namespace. Since enc.cpp includes abortStatus_t.h, the compiler will see the extern "C" qualified declaration of regular_abort and apply the extern "C" qualifier to the function definition as well.
If enc.cpp does not include abortStatus_t.h, there will be a link error since regular_abort will have C++ linkage.